src/main/scala/org/apache/spark/sql/delta/analysis/DeltaAnalysis.scala

```diff
@@ -47,8 +53,18 @@ case object DeltaAnalysis extends Rule[LogicalPlan] {
     case _: AlterTable => return plan
 
     // INSERT INTO by ordinal
-    case a @ AppendData(DataSourceV2Relation(d: DeltaTableV2, _, _), query, false)
-        if query.resolved && needsSchemaAdjustment(query, d.schema()) =>
+    case a @ AppendData(DataSourceV2Relation(d: DeltaTableV2, _, _), query, _, false)
+      if query.resolved && needsSchemaAdjustment(d.name(), query, d.schema()) =>
+      val projection = normalizeQueryColumns(query, d)
```

Comment:
If this logic is likely to need updating in the future, probably we want a helper method to share it between the INSERT cases. (Otherwise it's pretty short so I'm fine leaving it as is.)